src/common/meta/src/kv_backend/postgres.rs

```
@@ -0,0 +1,619 @@
+                let value: String = r.get(1);
+                KeyValue {
+                    key: key.into_bytes(),
+                    value: value.into_bytes(),
+                }
+            })
+            .collect();
+        Ok(BatchDeleteResponse { prev_kvs: kvs })
+    }
+
+    async fn compare_and_put(&self, req: CompareAndPutRequest) -> Result<CompareAndPutResponse> {
+        let key = process_bytes(&req.key, "CASKey")?;
+        let value = process_bytes(&req.value, "CASValue")?;
+        if req.expect.is_empty() {
+            let put_res = self.put_if_not_exists(key, value).await?;
+            return Ok(CompareAndPutResponse {
+                success: put_res,
+                prev_kv: None,
+            });
+        }
+        let expect = process_bytes(&req.expect, "CASExpect")?;
+
+        let res = self
+            .client
+            .query(CAS, &[&key, &value, &expect])
+            .await
+            .context(PostgresFailedSnafu)?;
+        match res.is_empty() {
+            true => Ok(CompareAndPutResponse {
+                success: false,
+                prev_kv: None,
+            }),
+            false => {
+                let mut kvs: Vec<KeyValue> = res
+                    .into_iter()
+                    .map(|r| {
+                        let key: String = r.get(0);
+                        let value: String = r.get(1);
+                        KeyValue {
+                            key: key.into_bytes(),
+                            value: value.into_bytes(),
+                        }
+                    })
+                    .collect();
+                Ok(CompareAndPutResponse {
+                    success: true,
+                    prev_kv: Some(kvs.remove(0)),
+                })
+            }
+        }
+    }
+}
+
+#[async_trait::async_trait]
+impl TxnService for PgStore {
+    type Error = Error;
+
+    async fn txn(&self, _txn: KvTxn) -> Result<KvTxnResponse> {
+        unimplemented!()
```

Better to return an error, such as `UnspportedOperationSnafu{}.fail()` etc.

---

It's better to leave a TODO here.

---

The metasrv relies on transactions, it's fine to implement this in future PRs